odoo/fields.py

```diff
@@ -2754,20 +2758,32 @@ def convert_to_display_name(self, value, record):
         return value.display_name if value else False
 
 
-class _Relational(Field):
+class _Relational(Generic[_T], Field[_T]):
     """ Abstract class for relational fields. """
     relational = True
     domain = []                         # domain for searching values
     context = {}                        # context for searching values
     check_company = False
 
-    def __get__(self, records, owner):
+    def __get__(self, records, owner) -> _T:
         # base case: do the regular access
         if records is None or len(records._ids) <= 1:
             return super().__get__(records, owner)
         # multirecord case: use mapped
         return self.mapped(records)
 
+    def setup(self, model):
+        if not self.comodel_name:
+            # Auto initialize the comodel name from the generic type's var if used
+            # This allows to get the comodel from the type when the field is declared
+            # as `partner = fields.Many2one[Partner]()`
```

Comment:
@rco-odoo But unfortunately there are some limitations.... https://github.com/acsone/odoo/pull/70

Reply:
Indeed, that's a limitation.

But consider this: what if class name = model name = SQL table name?  Basically the problem goes away, since you can use a string as a forward reference, and that string would be the model name.

Possibly I am thinking too wildly, and this is never going to happen.  But this would simplify tons of code and make the framework much easier to learn, wouldn't it?

Reply:
Or auto-converting CamelCase to dot.case, yes that makes sense to me. At the very least a convention to name classes will become necessary, if only to make it easy to import them. And you suggested recommending to make them available in odoo.addons.{addon}.models which is appealing too.